qiskit/circuit/tools/equivalence_checker/base_equivalence_checker.py

```diff
@@ -0,0 +1,64 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Model for Equivalence Checker Results."""
+
+from abc import ABC, abstractmethod
+from time import time
+
+
+class EquivalenceCheckerResult:
+    """Model for Equivalence Checker Results.
+
+    Attributes:
+        success (bool): True if the equivalence check terminated without
+            encountering an error.
+        equivalent (bool): True if the circuits are equivalent, and False
+            otherwise. Equals to None if `success` is False.
+        error_msg (str): An error message, when failing to perform the
+            check. Equals to None if `success` is True.
+        time_taken (float): run time of the equivalence checker.
+        Additional attributes can be added by the individual checkers.
+    """
+
+    def __init__(self, success, equivalent, error_msg):
```

Comment:
This is true. However since we inherit now from AlgorithmResult`, we stick to its framework. It would be a good idea to make `AlgorithmResult` inherit from `dict`, as in `OptimizeResult`.

Reply:
AlgorithmResult used to inherit from dictionary and was a dictionary in the past - look at the present Aqua version in comparison. When it was moved to Terra it was requested to change that aspect and have the getter/setter properties run off instance variables directly rather than entries from the dictionary and remove the hybrid aspect.